controllers/configuration_controller_test.go

```diff
@@ -2371,10 +2371,36 @@ func TestCheckWhetherConfigurationChanges(t *testing.T) {
 					Namespace:           "b",
 					ControllerNamespace: "b",
 				},
-				configurationType: "xxx",
+				configurationType: "HCL",
+			},
+			want: want{
+				errMsg: "",
+			},
+		},
+		"configuration type is remote": {
+			args: args{
+				meta: &TFConfigurationMeta{
+					ConfigurationCMName: "aaa",
+					Namespace:           "b",
+					ControllerNamespace: "b",
+				},
+				configurationType: "Remote",
 			},
 			want: want{
-				errMsg: "not found",
+				errMsg: "",
+			},
+		},
+		"create configuration for the first time": {
+			args: args{
+				meta: &TFConfigurationMeta{
+					ConfigurationCMName: "aa",
+					Namespace:           "b",
+					ControllerNamespace: "b",
+				},
+				configurationType: "HCL",
+			},
+			want: want{
+				errMsg: "",
```

Comment:
Case "Configuration is changed" is not tested, which is the point of this PR. Can you add some test for that?

Reply:
> Case "Configuration is changed" is not tested, which is the point of this PR. Can you add some test for that?

A case "Configuration changed, and reconcile"" was added